source/windows/secure_channel_tls_handler.c

```diff
@@ -124,6 +142,49 @@ static size_t s_message_overhead(struct aws_channel_handler *handler) {
     return sc_handler->stream_sizes.cbTrailer + sc_handler->stream_sizes.cbHeader;
 }
 
+/* Checks whether current system is running Windows 10 version `build_number` or later. This check is used
+   to determin availability of TLS 1.3. This will continue to be a valid check in Windows 11 and later as the
+   build number continues to increment upwards. e.g. Windows 11 starts at version 21H2 (build 22_000) */
+static bool s_is_windows_equal_or_above_version(DWORD build_number) {

```

Comment:
Trivial: If this function is only used to check against this one specific build and in one place, do we need an argument to be provided instead of just checking against WINDOWS_BUILD_1809? 

Also, the log feels a bit misleading. They all imply that the system is windows 10. The fail states should state that we're "Running windows version before build %u" and not "Running earlier windows 10 version before build %u". We can state "Running windows version before windows 10 build %u" if we know we are always checking against WINDOWS_BUILD_1809 which will always be windows 10.

Reply:
We have another similar check for ALPN (see a [comment from Bret](https://github.com/awslabs/aws-c-io/pull/641#discussion_r1610568909) in the previous PR). That's where this arg came from.
But ALPN check is quite different, so it's definitely shouldn't be changed here, as this PR is already quite big.

I'll fix the log message, but will keep the arg, in case we'll decide to reuse this function in ALPN check.